subtrack.Tests/SubscriptionCalculatorTests/GetNextPaymentDateTests.cs

```diff
@@ -2,45 +2,63 @@
 using subtrack.DAL.Entities;
 using subtrack.MAUI.Services;
 using subtrack.MAUI.Services.Abstractions;
+using System.Collections;
 
-namespace subtrack.Tests.SubscriptionCalculatorTests
+namespace subtrack.Tests.SubscriptionCalculatorTests;
+
+public class GetNextPaymentDateTests
 {
-    public class GetNextPaymentDateTests
+    private readonly ISubscriptionsCalculator _sut;
+    private readonly IDateTimeProvider _dateTimeProvider = Substitute.For<IDateTimeProvider>();
+
+    public GetNextPaymentDateTests()
+    {
+        _sut = new SubscriptionsCalculator(_dateTimeProvider);
+    }
+
+    [Fact]
+    public void GetNextPaymentDate_ReturnsCorrectNextPaymentDate()
     {
-        private readonly ISubscriptionsCalculator _sut;
-        private readonly IDateTimeProvider _dateTimeProvider = Substitute.For<IDateTimeProvider>();
-
-        public GetNextPaymentDateTests()
-        {
-            _sut = new SubscriptionsCalculator(_dateTimeProvider);
-        }
-
-        [Fact]
-        public void GetNextPaymentDate_ReturnsCorrectNextPaymentDate()
-        {
-            // Arrange
-            var subscription = new Subscription { LastPayment = new DateTime(2023, 06, 02) };
-            var expected = new DateTime(2023, 07, 02);
-
-            // Act
-            var result = _sut.GetNextPaymentDate(subscription);
-
-            // Assert
-            Assert.Equal(expected, result);
-        }
-
-        [Fact]
-        public void GetNextPaymentDate_WhenLastPaymentIsLastDayOfMonth_ReturnsPaymentDateWithLastDayOfNextMonth()
-        {
-            // Arrange
-            var subscription = new Subscription { LastPayment = new DateTime(2023, 05, 31) };
-            var expected = new DateTime(2023, 06, 30);
-
-            // Act
-            var result = _sut.GetNextPaymentDate(subscription);
-
-            // Assert
-            Assert.Equal(expected, result);
-        }
+        // Arrange
+        var subscription = new Subscription { LastPayment = new DateTime(2023, 6, 2) , FirstPaymentDay = new DateTime(2023, 6, 2).Day};
+        var expected = new DateTime(2023, 7, 2);
+
+        // Act
+        var result = _sut.GetNextPaymentDate(subscription);
+
+        // Assert
+        Assert.Equal(expected, result);
     }
+
+    [Theory]
+    [ClassData(typeof(GetNextPaymentDateTestData))]
+    public void GetNextPaymentDate_WhenLastPaymentIsLastDayOfMonth_ReturnsPaymentDateWithLastDayOfNextMonth(DateTime lastPayment, DateTime expectedNextPaymentDate)
```

Comment:
We still need to test the scenario described in the bug issue. Add firstPaymentDay to the test objects alternatively make a test case that specifically covers that "GetNext...ReturnsSameNextPaymentDateDay_AsFirstPaymentDay"